### PR TITLE
enables graph server for the menu

### PR DESCRIPTION
**Pull request by csell5**
adds a graph endpoint for the menu. Also adds a shim around the server to run the graph server independent of now.

# local updates needed
`npm i`

# running 
to run outside of now, run `npm run start:watch`.

### CODE REVIEW

File: graph/resolvers/queries/menu/menu.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  menu: (parent, args, { dataSources }) =>
+    Object.keys(menu)
+      .map(m => menu[m])
+      .reduce((acc, cur) => acc.concat(cur), []),
+  drinks: (parent, args, { dataSources }) => menu.drinks,
+  appetizers: (parent, args, { dataSources }) => menu.appetizers,
+  dinners: (parent, args, { dataSources }) => menu.dinners,
+  salads: (parent, args, { dataSources }) => menu.salads,
+  kids: (parent, args, { dataSources }) => menu.kids,
+  wraps: (parent, args, { dataSources }) => menu.wraps
```
Comment (saragibby) [inline graph/resolvers/queries/menu/menu.js]: do we need desserts here too?

File: graph/resolvers/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import queries from './queries';
+import mutations from './mutations';
+import fieldResolvers from './fieldResolvers';
+
+// interfaces
```
Comment (saragibby) [inline graph/resolvers/index.js]: comments baby 🙌

File: graph/typedefs/dataTypes/menuCategory.enum.graphql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+enum MenuCategory {
+  DRINK
+  APPETIZER
+  SALAD
+  DINNER_LANDFOOD
+  DINNER_SEAFOOD
+  DESERT
+  KIDS
+  WRAP
```
Comment (saragibby) [inline graph/typedefs/dataTypes/menuCategory.enum.graphql]: Looks like sandwiches are on the current menu, do we need to separate them out in the new menu?

File: graph/typedefs/queries/menu.graphql [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# import MenuItem from "../dataTypes/menuItem.graphql"
+
+type Query {
+  "query the entire menu"
+  menu: [MenuItem]
+
+  "qurey the drink menu"
```
Comment (saragibby) [inline graph/typedefs/queries/menu.graphql]: `qurey` -> `query`

File: graph/typedefs/dataTypes/menuCategory.enum.graphql [hunk 5]
```diff
@@ -0,0 +1,10 @@
+enum MenuCategory {
+  DRINK
+  APPETIZER
+  SALAD
+  DINNER_LANDFOOD
+  DINNER_SEAFOOD
+  DESERT
```
Comment (saragibby) [inline graph/typedefs/dataTypes/menuCategory.enum.graphql]: `DESERT` -> `DESSERT`

File: graph/resolvers/index.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import interfaces from './interfaces';
+
+// create the Apollo Server
```
Comment (saragibby) [inline graph/resolvers/index.js]: legit

File: graph/typedefs/queries/menu.graphql [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# import MenuItem from "../dataTypes/menuItem.graphql"
+
+type Query {
```
[the remark this answers is not in the document]
Comment (csell5) [reply graph/typedefs/queries/menu.graphql]: I cut them, as I couldn't find any desserts

File: graph/typedefs/dataTypes/menuCategory.enum.graphql (hunk 5 above)
Comment (csell5) [reply graph/typedefs/dataTypes/menuCategory.enum.graphql]: cut

File: graph/resolvers/queries/menu/menu.js (hunk 1 above)
Comment (csell5) [reply graph/resolvers/queries/menu/menu.js]: cut

File: graph/typedefs/queries/menu.graphql (hunk 4 above)
Comment (csell5) [reply graph/typedefs/queries/menu.graphql]: 👍

File: graph/typedefs/dataTypes/menuCategory.enum.graphql (hunk 3 above)
Comment (csell5) [reply graph/typedefs/dataTypes/menuCategory.enum.graphql]: I saw that too... I'm going to get this merged in and do a bit of refactoring on that. There are a few additional items which are slightly different, like beer, wine, that need to get added as well.

**Pull request merged**